pandas/src/inference.pyx

```diff
@@ -569,18 +569,17 @@ def maybe_convert_numeric(object[:] values, set na_values,
             seen_complex = True
         else:
             try:
-                status = floatify(val, &fval)
+                status = floatify(val, &fval, &maybe_int)
                 floats[i] = fval
                 if not seen_float:
-                    if '.' in val or fval == INF or fval == NEGINF:
-                        seen_float = True
-                    elif 'inf' in val:  # special case to handle +/-inf
-                        seen_float = True
-                    elif fval < fINT64_MAX and fval > fINT64_MIN:
-                        try:
-                            ints[i] = int(val)
-                        except ValueError:
-                            ints[i] = <int64_t> fval
+                    if maybe_int:
+                        if fval < fINT64_MAX and fval > fINT64_MIN:
+                            try:
+                                ints[i] = int(val)
+                            except ValueError:
```

Comment:
I believe this can still fail for an int > int64max (but less < than a uint64max)